libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs

```diff
@@ -11,6 +11,7 @@ public static class DialogExtensions
         public static async Task RunAsync(this Dialog dialog, ITurnContext turnContext, IStatePropertyAccessor<DialogState> accessor, CancellationToken cancellationToken)
         {
             var dialogSet = new DialogSet(accessor);
+            dialogSet.TelemetryClient = dialog.TelemetryClient;
```

Comment:
and your sure its not better to do this (conditionally perhaps) inside the Add method?